### ISSUE TITLE
[ListItem]: Keyboard navigation doesn't work if list item has tabbable children outside of the native list item 

### ISSUE THREAD

**Issue by swendlandt** [SAP/ui5-webcomponents#8391]
### Bug Description

We have implemented a custom UI5 web component extending the ListItemBase component. This component has tabbable elements which are not children of the native list item. This makes sense for example if you have a master detail layout, I can share more details about the use case directly.

In this case, you can not tab to the elements which are not children of the native list item. This is because in [shouldForwardTabAfter](https://github.com/SAP/ui5-webcomponents/blob/4e381e576492b9a9ee6a1fcb83f98e55f231016a/packages/main/src/ListItemBase.ts#L116) only the focus dom ref and its children are considered. If you fix that, you can not leave back from the nested elements, because the keydown handler is registered only on the native li. Then there is also another problem if you have a list as nested element because the focus gets stuck on the after element of the nested list.

This fix works in our case, but needs to be checked, especially the part with the after element is hacky: https://github.com/SAP/ui5-webcomponents/pull/8390

### Affected Component

ListItem

### Expected Behaviour

The elements inside the custom list item should be tabbable

### Isolated Example

https://github.com/swendlandt/example-list-item

### Steps to Reproduce

With the isolated example, try to tab through the list and its descendants.

### Priority

Medium

### UI5 Web Components Version

1.22.0

### Browser

Chrome, Edge, Firefox, Safari

### Operating System

MacOS

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by nnaydenow** [comment]
Hi @swendlandt,

I've extended the test page of your example of the list item as following:

```html
<ui5-button>Previous content</ui5-button>
<ui5-list mode="SingleSelect">
	<ui5-example-list-item text="master list item 1">
		<ui5-list id="childList" mode="SingleSelect">
			<ui5-li >Item 1</ui5-li>
			<ui5-li>Item 2</ui5-li>
			<ui5-li>Item 3</ui5-li>
		</ui5-list>
	</ui5-example-list-item>
	<ui5-example-list-item text="master list item 2">
		<ui5-link>test-link</ui5-link>
	</ui5-example-list-item>
</ui5-list>
<ui5-button>After content</ui5-button>
```

As per guideline the focus should be in following order:
1. Previous content button
2. First item (or previous focused element of list because here we save the focus on focusout) of the #parentList
3. First item (or previous focused element of list because here we save the focus on focusout) of the #childList
4. After content button

Can you share more about your use case and what are you trying to achieve? I'm curious why custom list item doesn't fit to you?

**Comment by swendlandt** [comment]
I will close this issue, because @nnaydenow proposed as another more straightforward solution restructuring our code, which worked just fine. Many thanks!
